services/api/src/resources/project/resolvers.ts

```diff
@@ -187,6 +187,47 @@ export const getProjectByName: ResolverFn = async (
   }
 };
 
+export const getProjectByOpenshift: ResolverFn = async (
+  root,
+  args,
+  {
+    sqlClient,
+    hasPermission,
+    models,
+    keycloakGrant,
+  },
+) => {
+  let where;
+  try {
+    await hasPermission('project', 'viewAll');
+  } catch (err) {
+    if (!keycloakGrant) {
+      logger.warn('No grant available for getProjectByOpenshift');
+      return [];
+    }
+  }
```

Comment:
This is not a sufficient way to verify the user is allowed to view the projects being returned. If the user has `viewAll` permissions, then we don't have to limit the SQL query. In the `else` case, we have to figure out what projects the user _does_ have access to, and limit the SQL query to only return those projects.

See [`getAllProjects`](https://github.com/amazeeio/lagoon/blob/main/services/api/src/resources/project/resolvers.ts#L50-L57) for an example.

Reply:
Thanks for the quick review. I've updated the PR accordingly. :)